Detections/3pNetworkDetetions/PhishiglinkExecutionObserved.yaml

```diff
@@ -0,0 +1,83 @@
+id: 2fed0668-6d43-4c78-87e6-510f96f12145
+name: Phishing link click observed in 3P Network device
+description: |
+  'sucessfull Phishing link accessed via "Palo Alto Networks","Fortinet","Check Point","Zscaler" devices.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId:  OfficeATP
+    dataTypes:
+    - OfficeATP
+    - Office365
+    - CommonSecurityLog
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CommandAndControl
+relevantTechniques:
+  - T1071
+query: |
+    //Finding security alerts and extracting the Entities 
+    SecurityAlert
+    //extracting Alert Entities 
+    | extend Entities = parse_json(Entities)
+    | mv-apply Entity = Entities on
+    (
+    where Entity.Type == 'account'
+    | extend EntityUPN = iff(isempty(Entity.UserPrincipalName), tostring(strcat(Entity.Name, "@", tostring (Entity.UPNSuffix))), tostring(Entity.UserPrincipalName))
+    )
+    | mv-apply Entity = Entities on
+    (
+    where Entity.Type == 'ip'|extend EntityIp = tostring(Entity.Address)
+    )
+    | mv-apply Entity = Entities on
+    (
+    where Entity.Type == 'host'| extend EntityHost = tostring(Entity.HostName)
+    )
+    | mv-apply Entity = Entities on
+    (
+    where Entity.Type == 'url'
+    | extend EntityUrl = tostring(Entity.Url)
+    )
+    | where not(ipv4_is_private(EntityIp))
+    | summarize
+    AccountUpn=take_any(EntityUPN),IPAddress=take_any(EntityIp),HostName=take_any(EntityHost),Url=take_any(EntityUrl),AlertTime= min(TimeGenerated)by SystemAlertId, ProductName
+    // filtering 3pnetwork devices 
+    | join kind= inner (CommonSecurityLog
+    | where DeviceVendor has_any  ("Palo Alto Networks", "Fortinet", "Check Point", "Zscaler")
```

Comment:
Given some of these vendors have multiple products do we also want to filter on product name?

Reply:
Up until now, we have noticed that this product is present in a majority of vendors. We have thoroughly examined the data schema. Moving forward, as we obtain additional product details and related data, we will analyze them and incorporate them into our list of products.